### ISSUE TITLE
Need a reliable way to describe list of packages required

### ISSUE THREAD

**Issue by onlyjsmith** [disarm-platform/fn-covariate-extractor#2]
@onlyjsmith commented on [Tue Dec 11 2018](https://github.com/disarm-platform/covariate_extractor/issues/3)

This is something R should be able to do, but I'm not certain how. 

Maybe through configuring it as a package? There's also [Packrat](https://rstudio.github.io/packrat/).

The desired outcome is: we have a simple way to recreate the list of packages required, without needing to go through every file to check for `library(...)` statements.

**Comment by onlyjsmith** [comment]
We've added the `install_packages.R` script - seems to do (some of) this job. The part it doesn't do is making sure that the _correct versions_ of packages are installed. Maybe there is a way to set that when you install a package?

**Comment by HughSt** [comment]
Yes I think this is possible

https://support.rstudio.com/hc/en-us/articles/219949047-Installing-older-versions-of-packages

But isn't this exactly what containerizing avoids? Or are we worried about situations where others build their own containers from our code?

**Comment by onlyjsmith** [comment]
Not exactly. The problem would be that `install.packages` will usually go get the most recent version of the package, and that version could change over time, potentially introducing backwards-incompatible changes: we won't have _completely_ reproducible builds. It would be the same for us or someone else building the image in the future.

I'd leave in icebox though. Let's wait till we hit this as a recurrent problem before fixing though, at the moment it's just a potential problem.

**Comment by HughSt** [comment]
Agreed
